lib/util/cache/repository/impl/client/local/index.ts

```diff
@@ -0,0 +1,35 @@
+import upath from 'upath';
+import { GlobalConfig } from '../../../../../../config/global';
+import { logger } from '../../../../../../logger';
+import { outputCacheFile, readCacheFile } from '../../../../../fs';
+import type { CacheClient, RepoCacheRecord } from '../../../types';
+
+export class LocalRepoCache implements CacheClient {
+  constructor(private platform: string, private repository: string) {}
```

Comment:
shouldn't those args passed to parent class as we need them on all implementations?

Reply:
now sure i follow, these are are passed and are available in the factory class. so it will be available for all implementations.

for example here is the factory in my S3 client implementation
https://github.com/Gabriel-Ladzaretti/renovate/blob/4a9e925f32341b76ac45b352586e737a03f513cc/lib/util/cache/repository/impl/client/cache-client-factory.ts#L8

Reply:
pass them to base class and make them protected properties

Reply:
`repository` already exists on at the `RepositoryCacheImpl` class, platform is available from the global config object.
should i add the platform explicitly even so?

Reply:
platform can probably fetched in contructor of base class from global config and made available as protected property. Or just fetch it from global config when it's needed.

Reply:
>Or just fetch it from global config when it's needed.

Ok, that's exactly what is done currently, just the repo prop is not protected instead of being private